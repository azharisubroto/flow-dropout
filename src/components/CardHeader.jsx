import PropTypes from 'prop-types'
import { useEffect, useState } from 'react'
import IconButton from '@/components/IconButton'
import { MdInfoOutline } from 'react-icons/md'
import ReactTooltip from 'react-tooltip'

const Card = ({ className, headline, tooltipText, children, ...other }) => {
  const [renderTooltip, setRenderTooltip] = useState(false)

  useEffect(() => {
    setRenderTooltip(true)
  }, [])

  return (
    <div className={`ra-card-header ${className ? className : ''}`} {...other}>
      {/* Headline and tooltip */}
      <div>
        <div className="d-flex">
          <div className="headline">{headline}</div>

          {/* info icon and tooltip */}
          {renderTooltip && renderTooltip && (
            <div className="icon">
              <IconButton size={22} data-for="info" data-tip={tooltipText}>
                <MdInfoOutline size={14} />
              </IconButton>
              <ReactTooltip id="info" place="top" type="dark" effect="solid" multiline={true} />
            </div>
          )}
        </div>
      </div>

      {/* Additional UI */}
      <div>{children}</div>

      <style jsx>{`
        .ra-card-header {
          border-bottom: solid 1px #ecf2ff;
          padding: 20px;
          display: flex;
          justify-content: space-between;
          align-items: center;
        }
        .d-flex {
          display: flex;
          flex-wrap: nowrap;
          align-items: center;
        }
        .headline {
          font-size: 18px;
          font-weight: normal;
          font-stretch: normal;
          font-style: normal;
          line-height: normal;
          letter-spacing: normal;
          color: #323a44;
        }
        .icon {
          margin-left: 10px;
        }
      `}</style>
    </div>
  )
}

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.any,
  headline: PropTypes.any,
  tooltipText: PropTypes.string
}

export default Card
