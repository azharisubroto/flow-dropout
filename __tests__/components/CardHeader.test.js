import React from 'react'
import { shallow } from 'enzyme'
import CardHeader from '@/components/CardHeader'

describe('<CardHeader />', () => {
  let wrapper, CardHeaderComp

  beforeEach(() => {
    CardHeaderComp = <CardHeader>My CardHeader</CardHeader>

    wrapper = shallow(CardHeaderComp)
  })

  it('should renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain contain correct class', () => {
    expect(wrapper.find('.ra-card-header')).toBeTruthy()
  })

  it('should contain contain .headline class', () => {
    expect(wrapper.find('.headline')).toBeTruthy()
  })
})
