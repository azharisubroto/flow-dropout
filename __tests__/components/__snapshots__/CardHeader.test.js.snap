// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardHeader /> should renders correctly 1`] = `ShallowWrapper {}`;
